specs/electra/beacon-chain.md

```diff
@@ -789,6 +841,56 @@ def slash_validator(state: BeaconState,
 
 ## Beacon chain state transition function
 
+### Execution engine
+
+#### Request data
+
+#### Engine APIs
+
+##### Modified `notify_new_payload`
+
+*Note*: The function `notify_new_payload` is modified to include the blob gas fee.
+
+```python
+def notify_new_payload(self: ExecutionEngine,
+                       execution_payload: ExecutionPayload,
+                       parent_beacon_block_root: Root,
+                       base_fee_per_blob_gas: uint64) -> bool:
+    """
+    Return ``True`` if and only if ``execution_payload`` is valid with respect to ``self.execution_state``.
+    """
+    ...
+```
+
+##### Modified `verify_and_notify_new_payload`
+
+```python
+def verify_and_notify_new_payload(self: ExecutionEngine,
+                                  state: BeaconState,
+                                  new_payload_request: NewPayloadRequest) -> bool:
+    """
+    Return ``True`` if and only if ``new_payload_request`` is valid with respect to ``self.execution_state``.
+    """
+    execution_payload = new_payload_request.execution_payload
+    parent_beacon_block_root = new_payload_request.parent_beacon_block_root
+
+    if not self.is_valid_block_hash(execution_payload, parent_beacon_block_root):
+        return False
+
+    if not self.is_valid_versioned_hashes(new_payload_request):
+        return False
+
+    # [Modified in Electra]
+    if not self.notify_new_payload(execution_payload,
+                                   parent_beacon_block_root,
+                                   get_base_fee_per_blob_gas(state)):
+        return False
+    
+    state.excess_blob_gas += execution_payload.blob_gas_used
```

Comment:
I think this like should be a part of the `process_execution_payload`. The other thing is that there is no excess blob gas computation, I believe it should be computed as follows:
```python
def compute_excess_blob_gas(excess_blob_gas: uint64, blob_gas_used: uint64) -> uint64:
    if blob_gas_used > TARGET_BLOB_GAS_PER_BLOCK:
        return excess_blob_gas + blob_gas_used - TARGET_BLOB_GAS_PER_BLOCK
    else:
        if excess_blob_gas < TARGET_BLOB_GAS_PER_BLOCK - blob_gas_used
            return uint64(0)
        else:
            return excess_blob_gas - (TARGET_BLOB_GAS_PER_BLOCK - blob_gas_used)
```